pkg/azure/actuator.go

```diff
@@ -440,41 +323,52 @@ func (a *Actuator) updateProviderStatus(ctx context.Context, logger log.FieldLog
 	return nil
 }
 
-func (a *Actuator) syncMint(ctx context.Context, cr *minterv1.CredentialsRequest, cloudCredsSecret *corev1.Secret, infraName string, infraResourceGroups []string, logger log.FieldLogger) error {
-	azureStatus, err := decodeProviderStatus(a.codec, cr)
-	if err != nil {
-		return err
+func copyCredentialsSecret(cr *minterv1.CredentialsRequest, src, dest *corev1.Secret) {
+	dest.ObjectMeta = metav1.ObjectMeta{
+		Name:      cr.Spec.SecretRef.Name,
+		Namespace: cr.Spec.SecretRef.Namespace,
+		Annotations: map[string]string{
+			minterv1.AnnotationCredentialsRequest: fmt.Sprintf("%s/%s", cr.Namespace, cr.Name),
+		},
 	}
-
-	azureSpec, err := decodeProviderSpec(a.codec, cr)
-	if err != nil {
-		return err
+	dest.Data = map[string][]byte{
+		AzureClientID:       src.Data[AzureClientID],
+		AzureClientSecret:   src.Data[AzureClientSecret],
+		AzureRegion:         src.Data[AzureRegion],
+		AzureResourceGroup:  src.Data[AzureResourceGroup],
+		AzureResourcePrefix: src.Data[AzureResourcePrefix],
+		AzureSubscriptionID: src.Data[AzureSubscriptionID],
+		AzureTenantID:       src.Data[AzureTenantID],
 	}
+}
 
-	if len(azureSpec.RoleBindings) == 0 {
-		msg := "No role specified in role bindings"
-		logger.Error(msg)
-		return &actuatoriface.ActuatorError{
-			ErrReason: minterv1.CredentialsProvisionFailure,
-			Message:   msg,
-		}
+func (a *Actuator) syncPassthrough(ctx context.Context, cr *minterv1.CredentialsRequest, cloudCredsSecret *corev1.Secret, logger log.FieldLogger) error {
+	syncErr := a.syncCredentialSecrets(ctx, cr, cloudCredsSecret, logger)
+
+	// Since we are live pivoting from Mint to Passthrough, try to clean up the old App Registration
+	cleanupErr := a.cleanupAfterPassthroughPivot(ctx, cr, cloudCredsSecret, logger)
+	if cleanupErr != nil {
+		logger.WithError(cleanupErr).Warn("unable to clean up previously minted App Regisration/Service Principal")
 	}
+	if syncErr == nil && cleanupErr != nil {
+		// A syncErr is more important to communicate about
+		// but do set a condition if the only issue was
+		// during cleanup.
+		return cleanupErr
+	}
+	return syncErr
+}
```

Comment:
This actually addresses the potential that we would fail to sync the secret, but still attempted a cleanup. That could lead to bad things. Good catch!